backend/src/main/java/com/redhat/cloud/notifications/routers/models/RequestEmailSubscriptionProperties.java

```diff
@@ -4,11 +4,13 @@
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
 
 import javax.validation.constraints.NotNull;
+import java.util.UUID;
 
 @JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
 public class RequestEmailSubscriptionProperties {
     @NotNull
     private boolean onlyAdmins;
+    private UUID groupId;
```

Comment:
Does this restrict the number of RBAC groups to one per endpoint? The images on Jira suggest that more than one group may be selected for an email endpoint.

Reply:
Yes, one per endpoint; multiple endpoints can be created (internally) to handle those combinations.

Reply:
So the frontend will call `POST /system/email_subscription` multiple times while the user is only adding one email integration? That sounds a bit weird. Is it impossible to handle several groups with a single email endpoint?

Reply:
This is done to allow (in the future) multiple configuration types and not making the endpoint type more complicated.
So one could do `send to this group, override user preferences and only admins`
```
{
  group: "some-id",
  sendToAdmins: true,
  overrideUserPreferences: true
}
```

Furthermore, each system/email_subscription are currently immutable and reused. So that when we create a behavior group with the action: "Send to all users". Only 1 endpoint is created and shared across all behavior groups that have the same action (send to all users).

The system endpoints are immutable. If we were to accept multiple groups in there, we could end with multiple combinations and maybe more complicated queries.

Having them as a single configuration for an email seemed more easier to manage and flexible to me.
See the [tests](https://github.com/RedHatInsights/notifications-backend/blob/master/backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java#L882) for info on how they behave.

Reply:
I remember the system endpoints immutability 😃 

Makes sense, thanks for clarifying!